front/src/modules/ui/modal/components/Modal.tsx

```diff
@@ -11,15 +11,39 @@ import {
 } from '@/ui/utilities/pointer-event/hooks/useListenClickOutside';
 
 import { ModalHotkeyScope } from './types/ModalHotkeyScope';
+import { ModalPadding, ModalSize } from './types/ModalTypes';
 
-const StyledModalDiv = styled(motion.div)`
+const StyledModalDiv = styled(motion.div)<{
+  size?: ModalSize;
+  padding?: ModalPadding;
+}>`
   display: flex;
   flex-direction: column;
   background: ${({ theme }) => theme.background.primary};
   border-radius: ${({ theme }) => theme.border.radius.md};
   overflow: hidden;
   max-height: 90vh;
   z-index: 10000; // should be higher than Backdrop's z-index
+
+  width: ${({ size, theme }) =>
+    size === 'small'
```

Comment:
@mustajab-ikram Rather than doing imbricated ternary, I think it's more readable using a `switch...case` when we have more than 2 possibilities :)

Reply:
yes, converted to switch case.